module/Application/tests/Model/Service/ServiceAbstractFactoryTest.php

```diff
@@ -0,0 +1,123 @@
+    public function services()
+    {
+        return [
+            [AccountCleanupService::class,
+                [
+                    UserRepositoryInterface::class => Mockery::mock(UserRepositoryInterface::class),
+                    ApplicationRepositoryInterface::class =>  Mockery::mock(ApplicationRepositoryInterface::class),
+                    'NotifyClient' => Mockery::mock(Client::class),
+                    'config' => [],
+                    UsersService::class =>  Mockery::mock(UsersService::class),
+                ]
+            ],
+            [PasswordService::class,
+                [
+                    UserRepositoryInterface::class => Mockery::mock(UserRepositoryInterface::class),
+                    AuthenticationService::class => Mockery::mock(AuthenticationService::class),
+                ]
+            ],
+            [PdfsService::class,
+                [
+                    ApplicationRepositoryInterface::class =>  Mockery::mock(ApplicationRepositoryInterface::class),
+                    'config' => [],
+                    'S3Client' => Mockery::mock(S3Client::class),
+                    'SqsClient' => Mockery::mock(SqsClient::class),
+                ]
+            ],
+            [SeedService::class,
+                [
+                    ApplicationRepositoryInterface::class =>  Mockery::mock(ApplicationRepositoryInterface::class),
+                    ApplicationsService::class => Mockery::mock(ApplicationsService::class),
+                ]
+            ],
+            [UsersService::class,
+                [
+                    LogRepositoryInterface::class => Mockery::mock(LogRepositoryInterface::class),
+                    UserRepositoryInterface::class => Mockery::mock(UserRepositoryInterface::class),
+                    ApplicationsService::class => Mockery::mock(ApplicationsService::class),
+                ]
+            ],
+            [ProcessingStatusService::class,
+                [
+                    HttpClient::class => Mockery::mock(HttpClient::class),
+                    'config' => ['processing-status' => ['endpoint' => 'test endpoint']]
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider services
+     * @param $service string Service class to check
+     */
+    public function testCanCreate($service)
+    {
+        $result = $this->factory->canCreate($this->container, $service);
+
+        $this->assertTrue($result);
+    }
+
+    /**
+     * @dataProvider services
+     * @param $service string Service class to check
+     * @param $dependancies array Dependencies that the mock container will provide
+     * @throws ApiProblemException
+     */
+    public function testInvoke($service, $dependancies)
+    {
+        foreach($dependancies as $key => $value){
+            $this->container->shouldReceive('get')->withArgs([$key])->once()->andReturn($value);
+        }
+
+        $this->factory->__invoke($this->container, $service);
+    }
+
```

Comment:
Trying something new with a test data provider, does this make sense to people or is it too difficult to work out what is going on?